pkg/config/mesh/kubemesh/watcher.go

```diff
@@ -43,11 +52,34 @@ func NewConfigMapWatcher(client kube.Client, namespace, name, key string) mesh.W
 
 	stop := make(chan struct{})
 	go c.Run(stop)
+
+	if extra != "" {
+		extrac := configmapwatcher.NewController(client, namespace, extra, func(cm *v1.ConfigMap) {
+			meshConfig := meshConfigMapData(cm, key)
+			w.HandleUserMeshConfig(meshConfig)
+		})
+
+		go extrac.Run(stop)
+	}
+
 	// Ensure the ConfigMap is initially loaded if present.
 	cache.WaitForCacheSync(stop, c.HasSynced)
```

Comment:
I think we need to wait for the sync of extra as well, otherwise we will have startup race conditions where user config is not yet read